### ISSUE TITLE
[FEA] Implement basic reductions for decimal columns

### ISSUE THREAD

**Issue by ChrisJar** [rapidsai/cudf#7503]
**Is your feature request related to a problem? Please describe.**
I wish to be able to perform basic reductions like `.sum()`, `.mean()`, or `.std()` on DataFrames and Series with `decimal` columns. 

**Describe the solution you'd like**
I would like to mimic the reductions available for `float` columns. For example:
```
df = cudf.DataFrame({'id': [0, 1, 1], 'val': [1.00, 1.01, 1.02]})
df.sum()
```
returns 
```
id     2.00
val    3.03
dtype: float64
```
but if the dataframe contains a `decimal` column
```
df['val'] = cudf.Series([decimal.Decimal(x) for x in [1.00, 1.01, 1.02]], dtype=cudf.Decimal64Dtype(7,3))
df.sum()
```
it returns
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-43-7e5fdb616c56> in <module>
----> 1 df.sum()

/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/cudf/core/dataframe.py in sum(self, axis, skipna, dtype, level, numeric_only, min_count, **kwargs)
   6063             numeric_only=numeric_only,
   6064             min_count=min_count,
-> 6065             **kwargs,
   6066         )
   6067 

/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/cudf/core/dataframe.py in _apply_support_method(self, method, axis, *args, **kwargs)
   6788             result = [
   6789                 getattr(self[col], method)(*args, **kwargs)
-> 6790                 for col in self._data.names
   6791             ]
   6792 

/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/cudf/core/dataframe.py in <listcomp>(.0)
   6788             result = [
   6789                 getattr(self[col], method)(*args, **kwargs)
-> 6790                 for col in self._data.names
   6791             ]
   6792 

/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/cudf/core/series.py in sum(self, axis, skipna, dtype, level, numeric_only, min_count, **kwargs)
   2985 
   2986         return self._column.sum(
-> 2987             skipna=skipna, dtype=dtype, min_count=min_count
   2988         )
   2989 

/home/u00u7rh1e72hXfsipJ357/miniconda3/envs/rapids-gpu-bdb/lib/python3.7/site-packages/cudf/core/column/column.py in sum(self, skipna, dtype, min_count)
   1257         self, skipna: bool = None, dtype: Dtype = None, min_count: int = 0
   1258     ):
-> 1259         raise TypeError(f"cannot perform sum with type {self.dtype}")
   1260 
   1261     def product(

TypeError: cannot perform sum with type decimal
```
The same error is returned for other basic reductions like `.mean()` and `.std()` as well.

**Comment by karthikeyann** [comment]
I can't reproduce this error. I get this error on `astype`.
`TypeError: Cannot interpret 'Decimal64Dtype(precision=7, scale=3)' as a data type`

can you give details on your environment?

Following code recreated the error in branch-0.19.
```
import decimal
import cudf
s = cudf.Series([decimal.Decimal(x) for x in [0.5, 1.5,2.5]], dtype=cudf.Decimal64Dtype(7,3))
s.sum()
```

**Comment by karthikeyann** [comment]
@codereport added decimal type reduction in libcudf. Python interface might be needed to utilize this.

**Comment by ChrisJar** [comment]
> I can't reproduce this error. I get this error on `astype`.
> `TypeError: Cannot interpret 'Decimal64Dtype(precision=7, scale=3)' as a data type`
> 
> can you give details on your environment?
> 
> Following code recreated the error in branch-0.19.
> 
> ```
> import decimal
> import cudf
> s = cudf.Series([decimal.Decimal(x) for x in [0.5, 1.5,2.5]], dtype=cudf.Decimal64Dtype(7,3))
> s.sum()
> ```

My apologies that astype call uses a PR that hasn't been merged yet. I'll edit the original post to your code

**Comment by codereport** [comment]
> @codereport added decimal type reduction in libcudf. Python interface might be needed to utilize this.

Correct. sum reductions are implemented, but don't have Python bindings yet. We only support a few binary operations atm.

**Comment by ChrisJar** [comment]
Closed with #7776
